Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -251,49 +251,70 @@ function New-LDAPIdentitySource {
 
 <#
     .Synopsis
-     Download the certificate from a domain controller
+     Download certificates from domain controllers and save them to local files
 #>
-function Get-CertificateFromDomainController {
+function Get-CertificateFromDomainContollerToLocalFile {
     param (
         [Parameter(
             Mandatory = $true)]
         [ValidateNotNull()]
-        [System.Uri]
-        $ParsedUrl,
-
-        [Parameter(
-            Mandatory = $true)]
-        [ValidateNotNull()]
-        [string]
-        $computerUrl
+        [string[]]
+        $remoteComputers
     )
-
-    try {
-        try {
-            $Command = 'nslookup ' + $ParsedUrl.Host + ' -type=soa'
-            $SSHRes = Invoke-SSHCommand -Command $Command -SSHSession $SSH_Sessions['VC'].Value
+    
+    $DestinationFileArray = @()
+    $exportFolder = "./"
+    foreach ($computerUrl in $remoteComputers) {
+        if (![uri]::IsWellFormedUriString($computerUrl, 'Absolute')) { 
+            throw "Incorrect Url format entered from: $computerUrl" 
         }
-        catch {
-            throw "The FQDN $($ParsedUrl.Host) cannot be resolved to an IP address. Make sure DNS is configured."
+        $ParsedUrl = [System.Uri]$computerUrl
+        if ($ParsedUrl.Port -lt 0 -OR $ParsedUrl.Host -eq "" -OR $ParsedUrl.Scheme -eq "") {
+            throw "Incorrect Url format entered from: $computerUrl" 
+        }
+        $ResultUrl = $ParsedUrl.Scheme + '://' + $ParsedUrl.Host + ':' + $ParsedUrl.Port
```

Comment:
There's an [existing method](https://learn.microsoft.com/en-us/dotnet/api/system.uri.getleftpart?view=net-8.0#remarks) for this.